File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxPathSum(self, root: Optional[TreeNode]) -> int:
        self.maxSum = float('-inf')

        def dfs(root)-> int:
            if not root:
                return 0
            
            leftmax = max(dfs(root.left),0)
            rightmax = max(dfs(root.right),0)
            curr_max = root.val + leftmax+ rightmax

            self.maxSum = max(curr_max,self.maxSum)
            return root.val + max(leftmax, rightmax)

        dfs(root)
        return self.maxSum

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {number} k
     * @return {number}
     */
    kthSmallest(root, k) {
        let s = [];
        let n=0;
        let curr = root;
        while(curr || s.length>0){
                while(curr){
                    s.push(curr);
                    curr= curr.left;
                }
                curr = s.pop();
                n++;
                if(n==k){
                    return curr.val;
                }
                curr = curr.right;
        }

    }
}
